### PR TITLE
Add Home Assistant.app v2020.7,1

### LINKED ISSUE #1037: Homebrew Cask support for MacOS
**Is your feature request related to a problem? Please describe.**

Homebrew is a really great tool to install software on MacOS and has support for GUI applications via `brew cask`. Being able to install Home Assistant Companion via Homebrew would be quite useful.

### LINKED ISSUE #1039: Make the macOS app available via `brew` package manager
**Is your feature request related to a problem? Please describe.**
This is not related to performance or bugs. This is purely a 'quality of life' request.

**Describe the solution you'd like**

I would like to see the macOS app distributed through the [homebrew package manager](https://brew.sh/). It's a wonderful package manager that makes is easy to install and - if the app allows it - update applications in a manor that's identical to the package managers from the *nix world.

I would like to be able to run:

```
me@some-host$ brew install home-assistant
```

and have the `home-assistant.app` fetched and put on my system. Ideally, any updates would also be installed with a `brew update`.

**Additional context**

the `brew` tool works by cloning git repos. Each repo has metadata about the application, where to fetch it from, how to validate it and anything else needed for automated install/remove/update.  For example, there is a CLI tool for working w/ the akamai APIs. The file that tells `brew` how to install / manage the `akamai` tool is [here](https://github.com/Homebrew/homebrew-core/blob/master/Formula/akamai.rb)

To [implement this request](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap), a few components need to be set up:

0. A repo that has the metadata set up for the applications in question. a *dedicated* repo is nice, but not required. This existing repo would work.

1. When a new macOS app is built, the repo containing release info needs to be updated. This process is relatively trivial to automate with github actions. There are _several_ implementations that can automate this process.

- https://github.com/marketplace/actions/homebrew-tap
- https://github.com/garden-io/update-homebrew-action
- https://github.com/marketplace/actions/bump-homebrew-formula

Happy to discuss further and answer any questions.  I'm not an osxDev so i'm a bit in. the dark about how the app is currently built and what the current approach is for determining when a release should be cut.  But once a release has been cut, it's easy to get that release added to brew.

### PR DESCRIPTION
**Pull request by ngdio**
Cask for the Home Assistant companion app. Fixes home-assistant/iOS#1037, fixes home-assistant/iOS#1039.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or **[documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version)**.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/home-assistant.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask "home-assistant" do
+  version "2020.7,1"
+  sha256 "118fb5a137784c0f546157a08d04d02925e94da0d804bd6aa17458a246df247a"
+
+  # github.com was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/home-assistant.rb:5]:
```suggestion
  # github.com/home-assistant/iOS/ was verified as official when first introduced to the cask
```

File: Casks/home-assistant.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  url "https://github.com/home-assistant/iOS/releases/download/mac/#{version.before_comma}/#{version.after_comma}/home-assistant-mac.zip"
+  appcast "https://github.com/home-assistant/iOS/releases.atom"
+  name "Home Assistant"
+  desc "Companion app for Home Assistant home automation software"
+  homepage "https://companion.home-assistant.io/"
+
+  app "Home Assistant.app"
```
Comment (vitorgalvao) [inline Casks/home-assistant.rb:12]: Please add a [`zap`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/zap.md).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Review by vitorgalvao — DISMISSED** [review]

**Review by vitorgalvao — APPROVED** [review]

**Comment by mweinelt** [comment]
This creates a mixup on repology, as this is not home-assistant but home-assistant companion.

https://repology.org/project/home-assistant/versions

Can you maybe rename this cask?

**Comment by vitorgalvao** [comment]
> Can you maybe rename this cask?

Not really something we want to do. We follow a strict token reference and by changing the casks name to be convenient to someone else it would be less convenient for us.

If you get upstream to change the app bundle’s name, the cask name will follow.

**Pull request merged**